src/System.Net.Http.WinHttpHandler/tests/WinHttpHandlerTest.cs

```diff
@@ -0,0 +1,965 @@
+
+            SendRequestHelper(
+                handler,
+                delegate
+                {
+                    handler.SslProtocols =
+                        SslProtocols.Tls |
+                        SslProtocols.Tls11 |
+                        SslProtocols.Tls12;
+                });
+
+            var expectedProtocols =
+                Interop.WinHttp.WINHTTP_FLAG_SECURE_PROTOCOL_TLS1 |
+                Interop.WinHttp.WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_1 |
+                Interop.WinHttp.WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2;
+            Assert.Equal(expectedProtocols, APICallHistory.WinHttpOptionSecureProtocols);
+        }
+
+        [Fact]
+        public async Task GetAsync_MultipleRequestsReusingSameClient_Success()
+        {
+            var handler = new WinHttpHandler();
+            var client = new HttpClient(handler);
+            TestServer.SetResponse(DecompressionMethods.None, ExpectedResponseBody);
+
+            var response = await client.GetAsync(FakeServerEndpoint);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            response = await client.GetAsync(FakeServerEndpoint);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            response = await client.GetAsync(FakeServerEndpoint);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            client.Dispose();
+        }
+
+        [Fact]
+        public async Task SendAsync_PostContentWithContentLengthAndChunkedEncodingHeaders_Success()
+        {
+            var handler = new WinHttpHandler();
+            var client = new HttpClient(handler);
+            client.DefaultRequestHeaders.TransferEncodingChunked = true;
+            TestServer.SetResponse(DecompressionMethods.None, ExpectedResponseBody);
+
+            var content = new StringContent(ExpectedResponseBody);
+            Assert.True(content.Headers.ContentLength.HasValue);
+            var request = new HttpRequestMessage(HttpMethod.Post, FakeServerEndpoint);
+            request.Content = content;
+
+            var response = await client.SendAsync(request);
+        }
+
+        [Fact]
+        public async Task SendAsync_PostNoContentObjectWithChunkedEncodingHeader_ExpectInvalidOperationException()
+        {
+            var handler = new WinHttpHandler();
+            var client = new HttpClient(handler);
+            client.DefaultRequestHeaders.TransferEncodingChunked = true;
+            TestServer.SetResponse(DecompressionMethods.None, ExpectedResponseBody);
+
+            var request = new HttpRequestMessage(HttpMethod.Post, FakeServerEndpoint);
+
+            await Assert.ThrowsAsync<InvalidOperationException>(async () => { var response = await client.SendAsync(request); });
+        }
+
+        // TODO: Need to skip this test due to missing native dependency clrcompression.dll.
+        // https://github.com/dotnet/corefx/issues/1298
+        [Fact(Skip="GitHub Issue #1298 - missing CLRCOMPRESSION.DLL")]
```

Comment:
@bartonjs Yes, thx.  When I first did this PR, 'ActiveIssue' wasn't working in the repro yet.  I will change to using that syntax in the next commit/update of this PR.